### PR TITLE
feat(log): Added output processing of the change log

### PR DESCRIPTION
**Pull request by shi9qiu**
Signed-off-by: Shi Qiu <shi1.qiu@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

Added the function of outputting the change log of project, component, and release in real-time.
You can edit "build-configuration\resources\log4j2.xml" for log settings (file name, location, rotation size, etc.).

> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)

Issue: #1339 

> * Did you add or update any new dependencies that are required for your change?

No.

Change some information of project, component, or release, then refer to the change log file for these change information.
The location of the log file is set in "build-configuration\resources\log4j2.xml".

> Have you implemented any additional tests?

No.

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java [hunk 1]
```diff
@@ -273,6 +273,15 @@ private void autosetReleaseClearingState(Release releaseAfter, Release releaseBe
     ////////////////////////////
     // GET INDIVIDUAL OBJECTS //
     ////////////////////////////
+    
+    public void addSelectLogs(Component component, User user){
```
Comment (ravi110336) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java:277]: Please check Indentation before { 
same for line 281

**Review by ravi110336 — COMMENTED** [review]
Hi @shi9qiu , Code looks good to me, just some minor review comments, I have tested the change log file it logs the changes which is made for the Project, Component and Release.

image

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/DatabaseHandlerUtil.java [hunk 2]
```diff
@@ -415,6 +416,35 @@ public static String convertObjectToJson(Object obj) {
...
 
+    /**
+     * Add Select Logs
+     * @param newDocVersion
+     * @param userEdited
+     * @param attachmentConnector
+     */
+    public static <T extends TBase> void addSelectLogs(T newDocVersion, String userEdited, AttachmentConnector attachmentConnector) {
+        Runnable changeLogRunnable = ()->{
+          try{
```
Comment (ravi110336) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/DatabaseHandlerUtil.java:427]: Indentation line 426,427.

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java [hunk 3]
```diff
@@ -265,6 +265,11 @@ public RequestStatus sendEmailForClearingRequestUpdate(ClearingRequest clearingR
     // GET INDIVIDUAL OBJECTS //
     ////////////////////////////
 
+    public void addSelectLogs(Project project, User user){
```
Comment (ravi110336) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java:268]: Indentation.

Comment (shi9qiu) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java:268]: Thank you for your review! I will fix them.

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java (hunk 1 above)
Comment (shi9qiu) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java]: Fixed it.

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java (hunk 3 above)
Comment (shi9qiu) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java]: Fixed it.

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/DatabaseHandlerUtil.java (hunk 2 above)
Comment (shi9qiu) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/DatabaseHandlerUtil.java]: Fixed it.

**Comment by shi9qiu** [comment]
Thanks for reviewing it. I have fixed them all. @ravi110336

**Comment by ravi110336** [comment]
Thanks @shi9qiu for incorporating the changes, code looks good to me,we can merge this PR.

**Pull request merged**